src/utils/plotSpecs/generateViolinSpec.js

```diff
@@ -419,8 +419,8 @@ const generateData = (
 ) => {
   const shouldBeDisplayed = (
     (cellId) => (
-      cellSetToDisplayId === 'All'
-      || cellSets.properties[cellSetToDisplayId].cellIds.has(cellId))
+      cellSets.properties[cellSetToDisplayId]?.cellIds.has(cellId))
+      || cellSetToDisplayId === 'All'
```

Comment:
This fixes the error with the violin plot. If the `cellSetToDisplayId` doesn't exist, the first clause will evaluate to null instead of throwin an error.